src/components/course/routes/tests/externalCourseEnrollmentLoader.test.jsx

```diff
@@ -0,0 +1,229 @@
+import { screen } from '@testing-library/react';
+import { when } from 'jest-when';
+import '@testing-library/jest-dom/extend-expect';
+
+import { renderWithRouterProvider } from '../../../../utils/tests';
+import makeExternalCourseEnrollmentLoader from '../externalCourseEnrollmentLoader';
+import {
+  extractEnterpriseCustomer,
+  getLateEnrollmentBufferDays,
+  queryCanRedeem,
+  queryCourseMetadata,
+  queryRedeemablePolicies,
+} from '../../../app/data';
+import { ensureAuthenticatedUser } from '../../../app/routes/data/utils';
+import { enterpriseCustomerFactory, authenticatedUserFactory } from '../../../app/data/services/data/__factories__';
+import { isDefinedAndNotNull } from '../../../../utils/common';
+
+jest.mock('../../../app/routes/data/utils', () => ({
+  ...jest.requireActual('../../../app/routes/data/utils'),
+  ensureAuthenticatedUser: jest.fn(),
+}));
+jest.mock('../../../app/data', () => ({
+  ...jest.requireActual('../../../app/data'),
+  extractEnterpriseCustomer: jest.fn(),
+  getLateEnrollmentBufferDays: jest.fn(),
+}));
+// jest.mock('@edx/frontend-platform/auth', () => ({
+//   ...jest.requireActual('@edx/frontend-platform/auth'),
+//   configure: jest.fn(),
+// }));
+// jest.mock('@edx/frontend-platform/logging', () => ({
+//   ...jest.requireActual('@edx/frontend-platform/logging'),
+//   configure: jest.fn(),
+//   getLoggingService: jest.fn(),
+// }));
```

Comment:
[nit] some commented out code

Reply:
Good catch! I recall intentionally leaving this commented when I copied it from another route loader, as I had a hunch it wasn't needed (it wasn't). I will clean up the other loader tests that have these same Jest mocks unnecessarily.